addons/api/addon/serializers/storage-bucket.js

```diff
@@ -8,6 +8,11 @@ import ApplicationSerializer from './application';
 export default class StorageBucketSerializer extends ApplicationSerializer {
   serialize(snapshot) {
     const serialized = super.serialize(...arguments);
+    //Dynamic credentials do not allow credential rotation, and the API requires this field to be
+    //sent along with the dynamic credential fields
+    if (serialized.attributes.role_arn) {
```

Comment:
FYI- for now, I have it here, but I will be refactoring these for the update functionality to separate the fields based on the type selection using `for` option in the generated model

Reply:
Can we do it like this? We're calling this credential type as "type" and switching off of it but this isn't really true is it?

We already have a type and it's `aws`, how would all of this work when we add another type like `azure`?

Reply:
good point, this may not work if we have more than one `type`. In that case, I could do something like how we do in [host-catalog ](https://github.com/hashicorp/boundary-ui/blob/main/addons/api/addon/serializers/host-catalog.js#L45)where we split the fields based on the record type and the credential type

Reply:
This has been refactored in [update](https://github.com/hashicorp/boundary-ui/pull/1901) pr